dali/pipeline/operators/geometric/bb_flip_test.cc

```diff
@@ -41,205 +49,124 @@ constexpr int kTestDataSize = 10;
+  int reference_data_idx = args["horizontal"] * 1 + args["vertical"] * 2;
 
-    auto batch_data = batch.template mutable_data<float>();
+  Roi anticipated_output_roi = Ltrb ? ltrb_wrapper.get(input_roi, reference_data_idx)
+                                    : wh_wrapper.get(input_roi, reference_data_idx);
 
-    InjectTestData<float, 0>(*test_data_, batch_data);
+  ASSERT_EQ(anticipated_output_roi.size(), output_roi.size())
+                        << "Inconsistent sizes (input vs output)";
+  for (size_t i = 0; i < output_roi.size(); i++) {
+    EXPECT_GT(kEpsilon, std::fabs(output_roi[i] - anticipated_output_roi[i]))
+                  << "Error exceeds allowed value";
   }
+}
 
+}  // namespace
 
-  const OpSpec GetOperatorSpec(bool ltrb_coordinates_type,
-                               bool vertical, bool horizontal) noexcept {
-    flip_type_vertical_ = vertical;
-    flip_type_horizontal_ = horizontal;
-    return OpSpec("BbFlip")
-            .AddArg("ltrb", ltrb_coordinates_type)
-            .AddArg("vertical", vertical ? 1 : 0)
-            .AddArg("horizontal", horizontal ? 1 : 0)
-            .AddInput("bb_input", "cpu")
-            .AddOutput("bb_output", "cpu");
+class BbFlipTest : public testing::DaliOperatorTest {
+  GraphDescr GenerateOperatorsGraph() const noexcept override {
+    GraphDescr graph("BbFlip");
+    return graph;
   }
-
-
- private:
-  const TestData *test_data_ = nullptr;
-  std::vector<std::vector<long int>> new_batch_size_;  //NOLINT
-  bool flip_type_vertical_, flip_type_horizontal_;
 };
 
-// XXX: `DALISingleOpTest` assumes, that input to the operator
-//      is always image and is templated by ImageType.
-//      Therefore this test had to be TYPED, regardless
-//      of the fact, that it's unnecessary.
-typedef ::testing::Types<RGB, BGR, Gray> Types;
-TYPED_TEST_CASE(BbFlipTest, Types);
-
-TYPED_TEST(BbFlipTest, VerticalWHTest) {
-  TensorList<CPUBackend> bb_test_data;
-  this->LoadBbData(bb_test_data, &wh_rois);
-  this->SetExternalInputs({std::make_pair("bb_input", &bb_test_data)});
-  this->RunOperator(this->GetOperatorSpec(false, true, false), .001);
-}
-
-
-TYPED_TEST(BbFlipTest, Vertical2PTest) {
-  TensorList<CPUBackend> bb_test_data;
-  this->LoadBbData(bb_test_data, &two_pt_rois);
-  this->SetExternalInputs({std::make_pair("bb_input", &bb_test_data)});
-  this->RunOperator(this->GetOperatorSpec(true, true, false), .001);
-}
+std::vector<Arguments> arguments = {
+        {{"horizontal", 1}, {"vertical", 0}},
+        {{"horizontal", 0}, {"vertical", 1}},
+        {{"horizontal", 0}, {"vertical", 0}},
+};
 
 
-TYPED_TEST(BbFlipTest, HorizontalWHTest) {
-  TensorList<CPUBackend> bb_test_data;
-  this->LoadBbData(bb_test_data, &wh_rois);
-  this->SetExternalInputs({std::make_pair("bb_input", &bb_test_data)});
-  this->RunOperator(this->GetOperatorSpec(false, false, true), .001);
+TEST_P(BbFlipTest, WhRoisTest) {
+  auto tlin = ToTensorList<CPUBackend>(rois_wh[0][0]);
```

Comment:
This only takes *one* bounding box. Why??

Reply:
Done